frontend/ui/detailview.jsx

```diff
@@ -594,42 +637,41 @@ class DetailViewComponent extends React.Component {
-                                {
-                                  (this.state.versionGrouping === 'version') ? (
-                                    <ul className="buildid-list">
-                                      {
-                                        item.subtitles.sort((a, b) => a < b).map(buildId => (
-                                          <dd
-                                            name={buildId}
-                                            className="buildid-link"
-                                            onClick={() => this.buildIdClicked(buildId)}
-                                            key={`buildid-${buildId}`}>
-                                            {buildId}
-                                          </dd>
-                                        ))
-                                      }
-                                    </ul>
-                                  ) : (
-                                    <p>
-                                      {item.subtitles[0]}
-                                    </p>
-                                  )
-                                }
-                              </small>
-                            </div>))
+                        legends.map(item => (
+                          <div key={item.title}>
+                            <Label check>
+                              <Input
+                                name={item.title}
+                                type="checkbox"
+                                checked={!this.state.disabledVersions.has(item.title)}
+                                onChange={this.versionCheckboxChanged} />
+                              {' '}
+                              {item.title}
+                            </Label>
+                            <small>
+                              {
+                                (this.state.versionGrouping === 'version') ? (
+                                  <ul className="buildid-list">
+                                    {
+                                      item.subtitles.sort((a, b) => a < b).map(buildId => (
+                                        <dd
+                                          name={buildId}
+                                          className="buildid-link"
+                                          onClick={() => this.buildIdClicked(buildId)}
+                                          key={`buildid-${buildId}`}>
+                                          {buildId}
+                                        </dd>
+                                      ))
+                                    }
+                                  </ul>
+                                ) : (
+                                  <p>
+                                    {item.subtitles[0]}
+                                  </p>
+                                )
+                              }
+                            </small>
+                          </div>))
```

Comment:
I think we can just revert this set of changes, once you calculate the versions to use directly.